File: gerenciador_tarefas/lib/models/tarefa.dart
import 'usuario.dart';

class Tarefa {
  final int? id;
  final String titulo;
  final String descricao;
  final Usuario? administrador;
  final List<Usuario> membros;
  final bool status;

  Tarefa({
    this.id,
    required this.titulo,
    required this.descricao,
    this.administrador,
    this.membros = const [],
    this.status = false,
  });

  factory Tarefa.fromJson(Map<String, dynamic> json) => Tarefa(
        id: json['id'],
        titulo: json['titulo'] ?? '',
        descricao: json['descricao'] ?? '',
        administrador: json['administrador'] != null ? Usuario.fromJson(json['administrador']) : null,
        membros: (json['membros'] as List<dynamic>?)?.map((e) => Usuario.fromJson(e)).toList() ?? [],
        status: json['status'] ?? false,
      );

  Map<String, dynamic> toJson() => {
        if (id != null) 'id': id,
        'titulo': titulo,
        'descricao': descricao,
        if (administrador != null) 'administrador': {'id': administrador!.id},
        'membros': membros.map((m) => {'id': m.id}).toList(),
        'status': status,
      };

  Tarefa copyWith({
    int? id,
    String? titulo,
    String? descricao,
    Usuario? administrador,
    List<Usuario>? membros,
    bool? status,
  }) =>
      Tarefa(
        id: id ?? this.id,
        titulo: titulo ?? this.titulo,
        descricao: descricao ?? this.descricao,
        administrador: administrador ?? this.administrador,
        membros: membros ?? this.membros,
        status: status ?? this.status,
      );
}

File: src/main/java/org/example/model/Tarefa.java
package org.example.model;
import jakarta.persistence.*;
import jdk.jfr.BooleanFlag;

import java.util.List;

@Entity
public class Tarefa {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String titulo;
    @Column(length = 1000)
    private String descricao;
    @ManyToOne
    private Usuario administrador;
    @ManyToMany
    private List<Usuario> membros;
    private boolean status;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public Usuario getAdministrador() {
        return administrador;
    }

    public void setAdministrador(Usuario administrador) {
        this.administrador = administrador;
    }

    public List<Usuario> getMembros() {
        return membros;
    }

    public void setMembros(List<Usuario> membros) {
        this.membros = membros;
    }

    public boolean isStatus() {
        return status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }
}
